### PR TITLE
raidboss: convert 03-hw/raid to TS

### PR DESCRIPTION
**Pull request by quisquous**

### CODE REVIEW

File: ui/raidboss/data/03-hw/raid/a8s.ts [hunk 1]
```diff
@@ -293,16 +317,18 @@ export default {
...
+      type: 'LosesEffect',
       netRegex: NetRegexes.losesEffect({ effectId: '400', capture: false }),
       run: (data) => delete data.lightning,
     },
     {
       id: 'A8S Compressed Lightning Soon',
+      type: 'GainsEffect',
       netRegex: NetRegexes.gainsEffect({ effectId: '400' }),
-      condition: (data) => data.lightning,
+      condition: (data) => data.lightning !== undefined,
```
Comment (panicstevenson) [inline ui/raidboss/data/03-hw/raid/a8s.ts:328]:
```suggestion
      condition: (data) => !!data.lightning,
```

File: ui/raidboss/data/03-hw/raid/a8s.ts [hunk 2]
```diff
@@ -820,11 +874,12 @@ export default {
     },
     {
       id: 'A8S Compressed Water Soon',
+      type: 'GainsEffect',
       netRegex: NetRegexes.gainsEffect({ effectId: '3FF' }),
-      condition: (data) => data.water,
+      condition: (data) => data.water !== undefined,
```
Comment (panicstevenson) [inline ui/raidboss/data/03-hw/raid/a8s.ts:879]:
```suggestion
      condition: (data) => !!data.water,
```

File: ui/raidboss/data/03-hw/raid/a10s.ts [hunk 3]
```diff
@@ -122,26 +134,29 @@ export default {
...
-        data.charges.push({
+        const charges = data.charges ??= [];
+        const chargeMap: { [abilityId: string]: string } = {
           '1AB8': 'getIn',
           '1AB9': 'getOut',
           '1ABA': 'spread',
           '1ABB': 'stackMarker',
-        }[matches.id]);
+        };
+        charges.push(chargeMap[matches.id] ?? 'unknown');
```
Comment (panicstevenson) [inline ui/raidboss/data/03-hw/raid/a10s.ts:144]: How does this get back to data.charges?

**Review by panicstevenson — APPROVED** [review]

File: ui/raidboss/data/03-hw/raid/a10s.ts (hunk 3 above)
Comment (quisquous) [reply ui/raidboss/data/03-hw/raid/a10s.ts:144]: My intention was that `charges === data.charges` here and so `charges.push` will change both of them.  Is that not true?

Comment (panicstevenson) [reply ui/raidboss/data/03-hw/raid/a10s.ts:144]: Oh, I misread line 137. Understood.

**Pull request merged**